### PR TITLE
Introducing GDPR compliance checks and policy

### LINKED ISSUE #9742: Update check: opt-out + preferences
Update check is currently always active. We need to

- opt-in (similar to telemetry)
- make it turn on/turn off in Preferences

### PR DESCRIPTION
**Pull request by calixtus**
Fixes #9742

These prefs were already there:
grafik
grafik

new:

grafik
grafik
grafik

### Mandatory checks
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by koppor** [comment]
Typical dialogs have this flow:

image

They hide the details.

In the settings, there are details:

image

(Source: https://www.nokia.com/)

Oh, and "Accept all" is really the term used there. 

Mercedes-Benz has following popup:

image

Settings scroll down:

image

Source: https://www.mercedes-benz.com/en/

---

Out of these two, I like the Nokia way the most.

Why not designing the dialog like the second one of Nokia?

- Dialog title "External services settings"
- Place the buttons on the left
- Can we also have first text and then the checkbox?
- Can we prepare an expansion of an item with details? Like links to the web search help etc?
- Button labels"Enable all", "Enable selected", "Disable all"

**Comment by calixtus** [comment]
As this is not a cookiebanner, I don't think that the examples really apply regarding the UX of a desktop application, but rather the already used design principles we use for dialogs in jabref. Meaning: checkbox on the left, buttons on the right.

Expandable checkboxes are nice, but imply that the dialog must be either very large to provide space for the items to expand and with a lot of empty space, if no item is expanded, or to grow and shrink the dialog with the items. You don't do that.

**Comment by koppor** [comment]
If consistency to the existing dialogs (designed more than 8 years ago) is imporant (which I agree, because we won't redesign the whole app), following wishlist items remain:

-  Dialog title "External services settings"
-  Button labels"Enable all", "Enable selected", "Disable all"

We could think about reducing the white space

image

Maybe with aligning the buttons in the middle (and moving the check boxes a bit to the right) - or put the buttons to the left, too.

**Comment by koppor** [comment]
I would rename "Web search for references" to "Web search for references and full texts". Alternatively, a second checkbox "Web search for full texts".

Side note: Maybe accept that PDF is the main file format and update the UI to use "Web search for PDFs". And replace in all other places "full text" by "PDFs".

**Comment by tobiasdiez** [comment]
Do we really have to show this dialog to every user? I don't know of any other software where one needs to explicitly accept version update checks or other accesses of web services. In particular for our own services (Update check and PDF grobid) I feel this is unnecessary, and for "web search" it is clear from the context that this accesses external services. 

I would prefer to have the content of this dialog moved to the preference window, so that users interested in their privacy can easily disable these services.

**Comment by calixtus** [comment]
This dialog should only be shown on the first run. And yes, according to the law, even before we send an IP address to the internet, we need to ask for opt in. This includes sadly also the version check.

**Comment by tobiasdiez** [comment]
If its only about the legal requirements, then we can put something like the following (from https://code.visualstudio.com/license)
>    Data Collection. The software may collect information about you and your use of the software, and send that to Microsoft. Microsoft may use this information to provide services and improve our products and services. You may opt-out of many of these scenarios, but not all, as described in the product documentation located at https://code.visualstudio.com/docs/supporting/faq#_how-to-disable-telemetry-reporting. There may also be some features in the software that may enable you and Microsoft to collect data from users of your applications. If you use these features, you must comply with applicable law, including providing appropriate notices to users of your applications together with Microsoft’s privacy statement. Our privacy statement is located at https://go.microsoft.com/fwlink/?LinkID=824704. You can learn more about data collection and use in the help documentation and our privacy statement. Your use of the software operates as your consent to these practices.
>    Processing of Personal Data. To the extent Microsoft is a processor or subprocessor of personal data in connection with the software, Microsoft makes the commitments in the European Union General Data Protection Regulation Terms of the Online Services Terms to all customers effective May 25, 2018, at https://docs.microsoft.com/legal/gdpr.

in our eula that users agree upon when installing jabref. This seems to be the standard way to handle this.

**Comment by calixtus** [comment]
I agree, that could be a solution. It's definitely an opt in solution. Yet, we don't have an eula, just the mit license when installing.

**Comment by tobiasdiez** [comment]
Should be easy to add via `license-path` option of jpackage: https://docs.oracle.com/en/java/javase/20/docs/specs/man/jpackage.html

**Comment by calixtus** [comment]
The technical solution is not the problem. It's the content of the eula, since we have no idea what to put in it. Any examples?

**Comment by tobiasdiez** [comment]
I did a bit of research but couldn't find a single open source software that requires a special-opt in for webservices. The most explicit is Firefox which link to their privacy policy under the download button (https://www.mozilla.org/en-US/firefox/channel/desktop/) and have a privacy policy link in their "About" dialog. Other examples such as owncloud don't show anything related to privacy in their installation or on-boarding process. It seems to be consensus that eula are not necessary for open source projects since the main license already specifies all necessary rights and has a warranty/liability clause.

So without being an expert on the legal side, it seems that there is no need for an explicit opt-in. Just add to our privacy policy that the desktop application access certain web services to provide additional features (similar to what I've cited above from the vscode eula) and link this privacy policy from a few places inside the software.

**Comment by calixtus** [comment]
### DevCall discussion
We agree that we want to avoid popups whenever possible. Introducing an eula seems to be a good idea. Drafting one and introducing it to the installer should be investigated.

File: PRIVACY.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+This privacy policy may be changed eventually. 
+We encourage you to check this Privacy Policy periodically for any changes.
+Any material change will be mentioned in the changelog and in our blog.
+
+This privacy policy is in effect as of the day mentioned as "last updated" above and will remain in effect except with respect to any changes in its provisions in the future, which will be in effect immediately after being posted on this page.
+
+## Contact
+For any questions or concerns regarding the privacy policy, please send us an email to <vorstand@jabref.org> or write to
+
+JabRef e.V.<br>
```
Comment (Siedlerchr) [inline PRIVACY.md:104]: <br> in markdown?

Comment (calixtus) [reply PRIVACY.md:104]: https://www.markdownguide.org/basic-syntax#line-break-best-practices

File: PRIVACY.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+- Anonymized statistical data on the use of the graphical user interface for internal analysis purposes (by default *disabled*).
+
+### Storing information
+JabRef does not store any personal information, except:
+- Your proxy username and password, if you decide to store them (by default *disabled*).
+- Any personal API key you use to access third party online services (by default *disabled*). 
+
+### Sharing information
+Certain operations you perform in JabRef may trigger requests to public third-party services such as Zotero, Crossref or the Library of Congress for metadata retrieval.
+These third parties may log besides your IP address and the search terms (e.g., DOI, ISBN or the current URL) according to their privacy policies.
```
Comment (Siedlerchr) [inline PRIVACY.md:31]:
```suggestion
These third parties may log additional information besides your IP address and the search terms (e.g., DOI, ISBN or the current URL) according to their privacy policies.
```

File: PRIVACY.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+| [The SAO/NASA Astrophysics Data System](https://ui.adsabs.harvard.edu/)                                         | <https://ui.adsabs.harvard.edu/help/privacy/> |
+| [Unpaywall](https://unpaywall.org/)                                                                             | <https://unpaywall.org/legal/privacy> |
+| [zbMATH Open](https://www.zbmath.org)                                                                           | <https://zbmath.org/privacy-policy/> |
+
+&ast; *Note: The Mr. DLib service is used for the related articles tab in the entry editor and collects also your language, your browser and operating system (by default *disabled*).* 
+
+## JabRef Browser Extension
+
+No personal data (like name, email address, billing address or credit card) is collected.
+By using the extension you will remain completely anonymous.
```
Comment (Siedlerchr) [inline PRIVACY.md:75]: I would remove this sentence

File: PRIVACY.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+### Collecting information
+
+When you access a website with your web browser, the Browser Extension is designed to save time by allowing you to quickly import bibliographic information into JabRef.
+Therefor the extension collects and analyzes some information concerning your browser activity.
```
Comment (Siedlerchr) [inline PRIVACY.md:80]: The extension does not really collect or store information. 
Suggestion: 
When actively used, the extension has access to the current website and its content. The information is only for the purpose of processing citations.

File: PRIVACY.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
+# Privacy Policy
+Last updated: 2023-08-24
+
+Your privacy is a fundamental right JabRef e.V. respects and supports.
+By using JabRef and its related online services, you choose to share some of this information.
+In this Privacy Policy We explain how We collect, use, and share information about you, along with the choices you have.
```
Comment (tobiasdiez) [inline PRIVACY.md:6]: We, Us and Privacy Policy should follow standard English capitalization rules (or at least be consistently written)

File: PRIVACY.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
+# Privacy Policy
+Last updated: 2023-08-24
+
+Your privacy is a fundamental right JabRef e.V. respects and supports.
+By using JabRef and its related online services, you choose to share some of this information.
```
Comment (tobiasdiez) [inline PRIVACY.md:5]: What refers 'this' to, reorder a bit here?

File: PRIVACY.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+### Collecting information
+JabRef does not collect any personal information directly linked to you.
+However, on certain occasions the software application will send some information to the online services of JabRef e.V.:
+- On application start, JabRef will check for the latest version online (by default *enabled*).
+- Information about a journal you are citing when looking for more information about this journal, using our journal database (by default *enabled*).
+- A pdf document you automatically want to extract citation information from, using our GROBID service (by default *disabled*).
+- Anonymized statistical data on the use of the graphical user interface for internal analysis purposes (by default *disabled*).
+
+### Storing information
+JabRef does not store any personal information, except:
```
Comment (tobiasdiez) [inline PRIVACY.md:25]:
```suggestion
JabRef only stores the following personal information locally on your computer:
```

File: PRIVACY.md (hunk 2 above)
Comment (tobiasdiez) [reply PRIVACY.md:31]: I would emphasize that we don't send any personal information to these services, except for what is strictly necessary to perform the task. In particular, only the ip Adress and the provided search terms are thansfered.

File: PRIVACY.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+## Contact
+For any questions or concerns regarding the privacy policy, please send us an email to <vorstand@jabref.org> or write to
+
+JabRef e.V.<br>
+Josef-Lanner-Str. 9<br>
+71069 Sindelfingen<br>
+Germany
+
+If you get in touch with us, We may aks you to provide Us with certain personal information (e.g. name and email address) that can be used to contact or identify you.
```
Comment (tobiasdiez) [inline PRIVACY.md:111]: That last sentence feels very much out of place here. Just delete it?

File: PRIVACY.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+By using JabRef and its related online services, you choose to share some of this information.
+In this Privacy Policy We explain how We collect, use, and share information about you, along with the choices you have.
+
+Please remember that no method of transmission over the Internet, or method of electronic storage is absolute secure.
+While We strive to use commercially acceptable means to protect your Personal Data, We cannot guarantee its absolute security.
+Also, whenever you communicate through the internet, your IP-Address will be transmitted and retained by other parties for technical reasons.
+
+The term 'Personal information' in this policy means any information that either directly identifies you or can be somehow linked to you. 'JabRef' is the software application, JabRef e.V. provides.
+
+## JabRef
```
Comment (tobiasdiez) [inline PRIVACY.md:14]:
```suggestion
## Desktop Application 
```

File: PRIVACY.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+Your privacy is a fundamental right JabRef e.V. respects and supports.
+By using JabRef and its related online services, you choose to share some of this information.
+In this Privacy Policy We explain how We collect, use, and share information about you, along with the choices you have.
+
+Please remember that no method of transmission over the Internet, or method of electronic storage is absolute secure.
+While We strive to use commercially acceptable means to protect your Personal Data, We cannot guarantee its absolute security.
+Also, whenever you communicate through the internet, your IP-Address will be transmitted and retained by other parties for technical reasons.
+
+The term 'Personal information' in this policy means any information that either directly identifies you or can be somehow linked to you. 'JabRef' is the software application, JabRef e.V. provides.
```
Comment (tobiasdiez) [inline PRIVACY.md:12]: Move the last sentence to the next subsection.

Comment (tobiasdiez) [inline PRIVACY.md:12]:
```suggestion
The term 'Personal information' in this policy means any information that either directly identifies you or can be somehow linked to you. 'JabRef' refers to the desktop application that is provided by JabRef e.V.
```

**Review by tobiasdiez — COMMENTED** [review]
Reads very nicely, good job! Only a few minor remarks, and please run a final spelling check over everything.

File: PRIVACY.md (hunk 5 above)
Comment (calixtus) [reply PRIVACY.md:6]: I believe "Privacy Policy" is a fixed term and a name, so I uppercased it in the document, but I'm not sure.

File: PRIVACY.md (hunk 8 above)
Comment (calixtus) [reply PRIVACY.md:111]: Should go somewhere, because that is in fact also an occasion, we collect and store data.

Comment (tobiasdiez) [reply PRIVACY.md:111]: Then it should be a complete new section...

**Comment by Siedlerchr** [comment]
grafik

**Pull request merged**